shell/browser/ui/file_dialog_gtk.cc

```diff
@@ -369,6 +369,24 @@ void FileChooserDialog::AddFilters(const Filters& filters) {
   }
 }
 
+bool CanPreview(const struct stat& st) {
+  // Only preview regular files; pipes may hang.
+  // See https://crbug.com/534754.
+  if (!S_ISREG(st.st_mode)) {
+    return false;
+  }
+
+  // Don't preview huge files; they may crash.
+  // https://github.com/electron/electron/issues/31630
+  // Setting an arbitrary filesize max t at 100 MB here.
+  constexpr off_t ArbitraryMax = 100000000ULL;
+  if (st.st_size >= ArbitraryMax) {
+    return false;
```

Comment:
tiny nit but we can probably just
```
return st.st_size < ArbitraryMax;
```
here!

Reply:
SGTM. 957804830ea4b1857dbc65e7e9f969008abdea16